pytorch_lightning/trainer/connectors/logger_connector/epoch_result_store.py

```diff
@@ -462,11 +388,9 @@ def update_logger_connector(self, fx_name: str = None) -> None:
         logger_connector.callback_metrics.pop("epoch", None)
 
     def run_batch_from_func_name(self, func_name) -> Dict:
-        results = []
-        for fx_name, hook_result in self._internals.items():
-            func = getattr(hook_result, func_name)
-            results.append(func(include_forked_originals=False))
-        return dict(ChainMap(*sum(results, [])))
+        results = [getattr(hook_result, func_name) for hook_result in self._internals.values()]
+        results = [func(include_forked_originals=False) for func in results]
+        return {k: v for d in sum(results, []) for k, v in d.items()}  # List[List[dict]] -> dict
```

Comment:
I'd assume a chain map to be more efficient than dict comprehensions, no?

Reply:
I have done some tests and ChainMap weren't faster. I am good with this 👍